ocaml/xapi/repository_helpers.ml

```diff
@@ -672,22 +673,16 @@ let eval_guidances ~updates_info ~updates ~kind ~livepatches ~failed_livepatches
       UpdateIdSet.empty lps
   in
   let upd_ids_of_livepatches = extract_upd_ids livepatches in
-  let upd_ids_of_failed_livepatches = extract_upd_ids failed_livepatches in
   List.fold_left
     (fun acc u ->
-      match
-        eval_guidance_for_one_update ~updates_info ~update:u ~kind
-          ~upd_ids_of_livepatches ~upd_ids_of_failed_livepatches
-      with
-      | Some g ->
-          GuidanceSet.add g acc
-      | None ->
-          acc
+      eval_guidance_for_one_update ~updates_info ~update:u ~kind
+        ~upd_ids_of_livepatches
+      |> GuidanceSet.of_list
+      |> GuidanceSet.union acc
     )
     GuidanceSet.empty updates
   |> append_livepatch_guidances ~updates_info ~upd_ids_of_livepatches
```

Comment:
Sorry, I cannot understand the logic here, I see that the livepatch guidances have been taking into account in `eval_guidance_for_one_update` for `recommended` guidance, but why we need to `append_livepatch_guidances ` for every `kind` of guidance again?
Would you please help to explain it a bit, Thank you!

Reply:
This is for the stand-alone livepatches. RPM packages have been updated but livepatches are still available.
```
 622 (* In case that the RPM in an update has been installed (including livepatch file),
 623  * but the livepatch has not been applied.
 624  * In other words, this RPM update will not appear in parameter [updates] of
 625  * function [eval_guidances], but the livepatch in it is still applicable.
 626  *)
```

Reply:
Yeah. This is not correct. Fixed it up.